awx_collection/test/awx/test_completeness.py

```diff
@@ -0,0 +1,251 @@
+
+# Read-only endpoints are dynamically created by an options page with no POST section.
+# Noramlly a read-only endpoint should not have a module (i.e. /api/v2/me) but sometimes we reuse a name
+# For example, we have a tower_role module but /api/v2/roles is a read only endpoint.
+# Rhis list indicates which read-only endpoints have associated modules with them.
+read_only_endpoints_with_modules = ['tower_settings', 'tower_role']
+
+# If a module should not be created for an endpoint and the endpoint is not read-only add it here
+# THINK HARD ABOUT DOING THIS
+no_module_for_endpoint = []
+
+# Some modules work on the related fields of an endpoint. These modules will not have an auto-associated endpoint
+no_endpoint_for_module = [
+    'tower_import', 'tower_meta', 'tower_export', 'tower_job_launch', 'tower_job_wait', 'tower_job_list',
+    'tower_license', 'tower_ping', 'tower_receive', 'tower_send', 'tower_workflow_launch', 'tower_job_cancel',
+    'tower_workflow_template',
+]
+
+# Global module parameters we can ignore
+ignore_parameters = [
+    'state', 'new_name',
+    # The way we implemented notifications was to add these fileds to modules which can notify
+    'notification_templates_approvals',
+    'notification_templates_error',
+    'notification_templates_started',
+    'notification_templates_success',
+]
+
+# Some modules take additional parameters that do not appear in the API
+# Add the module name as the key with the value being the list of params to ignore
+no_api_parameter_ok = {
+    # The credential module used to take a lot of parameters which are now combined into "inputs"
+    'tower_credential': [
+        'authorize', 'authorize_password', 'become_method', 'become_password', 'become_username', 'client',
+        'domain', 'host', 'kind', 'password', 'project', 'secret', 'security_token', 'ssh_key_data',
+        'ssh_key_unlock', 'subscription', 'tenant', 'username', 'vault_id', 'vault_password',
+    ],
+    # The wait is for wheter or not to wait for a projec update on change
+    'tower_project': ['wait'],
+    # Existing_token and id are for working with an exisitng tokens
+    'tower_token': ['existing_token', 'existing_token_id'],
+    # Children and hosts are how tower_group deals with associations
+    'tower_group': ['children', 'hosts'],
+    # Credential and Vault Credential are a legacy fields for old Towers
+    # Credentials/labels/survey spec is now how we handle associations
+    # We take an organization here to help with the lookups only
+    'tower_job_template': ['credential', 'vault_credential', 'credentials', 'labels', 'survey_spec', 'organization'],
+    # Always_nodes, failure_nodes, success_nodes, credentials is how we handle associations
+    # Organization is how we looking job templates
+    'tower_workflow_job_template_node': ['always_nodes', 'failure_nodes', 'success_nodes', 'credentials', 'organization'],
```

Comment:
Could we also do some general solution (via parameter help text) for related fields?

People are general averse to hard lists of fields, and in cases like these, we can get real value added for the user by putting this signals in a standard form in the help text in DOCUMENTATION.